nemo_skills/pipeline/utils/commands.py

```diff
@@ -0,0 +1,139 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Minimal command builders for declarative pipeline interface.
+
+These are thin wrappers around existing command construction utilities
+in server.py and other modules, designed to work with Command objects.
+"""
+
+from typing import Dict, Optional, Tuple
+
+from nemo_skills.pipeline.utils.exp import get_sandbox_command
+from nemo_skills.pipeline.utils.server import get_free_port, get_server_command
+
+
+def vllm_server_command(
+    cluster_config: Dict,
+    model: str,
+    port: Optional[int] = None,
+    server_type: str = "vllm",
+    gpus: int = 8,
+    nodes: int = 1,
+    args: str = "",
+    entrypoint: Optional[str] = None,
+    **kwargs,
+) -> Tuple[str, Dict]:
+    """Build vLLM server command.
+
+    Args:
+        cluster_config: Cluster configuration dictionary
+        model: Model path or name
+        port: Port to use (if None, will use get_free_port)
+        server_type: Type of server (vllm, sglang, trtllm, megatron)
+        gpus: Number of GPUs
+        nodes: Number of nodes
+        args: Additional server arguments
+        entrypoint: Custom entrypoint script
+
+    Returns:
+        Tuple of (command_string, metadata_dict)
+    """
+    if port is None:
+        port = get_free_port(strategy="random")
+
+    cmd, num_tasks = get_server_command(
```

Comment:
While we are at this, can we now isolate all logic in `get_server_command` into this function. In the other function, can we throw a warning,
```
import warnings
warnings.warn("deprecation message", DeprecationWarning)
```

In `get_server_command`, we can then just return the `cmd` from this function. 

I think we'll need to do this to slowly transition away from old designed functions, and keep track on things that need to be updated or removed.